src/workerd/api/pyodide/setup-emscripten.c++

```diff
@@ -0,0 +1,105 @@
+#include "setup-emscripten.h"
+
+#include "workerd/util/autogate.h"
+
+#include <workerd/io/trace.h>
+#include <workerd/io/worker.h>
+
+#include <pyodide/pyodide.capnp.h>
+namespace workerd::api::pyodide {
+
+v8::Local<v8::Module> loadEmscriptenSetupModule(
+    jsg::Lock& js, capnp::Data::Reader emsciptenSetupJsReader) {
+  v8::Local<v8::String> contentStr = jsg::v8Str(js.v8Isolate, emsciptenSetupJsReader.asChars());
+  v8::ScriptOrigin origin(
+      jsg::v8StrIntern(js.v8Isolate, "pyodide-internal:generated/emscriptenSetup"), 0, 0, false, -1,
+      {}, false, false, true);
+  v8::ScriptCompiler::Source source(contentStr, origin);
+  return jsg::check(v8::ScriptCompiler::CompileModule(js.v8Isolate, &source));
+}
+
+jsg::JsValue resolvePromise(jsg::Lock& js, jsg::JsValue prom) {
+  auto promise = KJ_ASSERT_NONNULL(prom.tryCast<jsg::JsPromise>());
+  if (promise.state() == jsg::PromiseState::PENDING) {
+    js.runMicrotasks();
```

Comment:
Can doing this block the thread for too long and prevent other isolates from doing work?

I guess we will run this in its own thread, so this won't be an issue?

Reply:
Previously this logic would run in the global await that the loading of the pyodide bundle would block on. It's essentially the exact same logic just moved.
This means that we aren't blocking for any more time than we previously did but simply blocking at stage that's a bit earlier.

For workerd that doesn't matter at all.
For our internal use case it will maintain the original behaviour for any isolate that isn't part of the pool (yeah those still exist).
If the isolate is part of the pool then this will run in a separate thread and actually improve the current status.